### PR TITLE
arrays / it-academy.by / M-JC1-69-21 / Bozhen Davydov / Done tasks 1-3

### PR DESCRIPTION
**Pull request by davydovbv**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/array/Task02.java [hunk 1]
```diff
@@ -31,9 +34,15 @@ public static void main(String[] args) {
...
-        return 0.0;
+        double summ = 0.0;
+        int quantityMarks = 0;
+        for (int[] mark : marks) {
+            for (int i : mark) {
+                summ += i;
+            }
+            quantityMarks += mark.length;
+        }
+        return BigDecimal.valueOf(summ / quantityMarks).setScale(2, RoundingMode.HALF_UP).doubleValue();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task02.java:45]:
```suggestion
        return BigDecimal.valueOf(summ / (marks.length * marks[0].length)).setScale(2, RoundingMode.HALF_UP).doubleValue();
```

File: src/main/java/com/rakovets/course/java/core/practice/array/Task02.java [hunk 2]
```diff
@@ -43,9 +52,15 @@ static double getAverageMark(int[][] marks) {
      * @return минимальная отметка
      */
     static int getMinMark(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+      int minMark = marks [0][0];
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task02.java:55]: Violation of Code Convention

File: src/main/java/com/rakovets/course/java/core/practice/array/Task02.java [hunk 3]
```diff
@@ -55,8 +70,14 @@ static int getMinMark(int[][] marks) {
      * @return максимальная отметка
      */
     static int getMaxMark(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int maxMark = marks [0][0];
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task02.java:73]:
```suggestion
        int maxMark = marks[0][0];
```

File: src/main/java/com/rakovets/course/java/core/practice/array/Task03.java [hunk 4]
```diff
@@ -33,9 +35,17 @@ public static void main(String[] args) {
...
-        return null;
+        double[] averageMarks = new double[marks.length];
+        for (int i = 0; i < marks.length; i++) {
+            double summ = 0.0;
+            int quantityMarks = 0;
+            for (int j = 0; j < marks[i].length; j++) {
+                summ += marks[i][j];
+                quantityMarks ++;
+            }
+            averageMarks[i] = BigDecimal.valueOf(summ / quantityMarks).setScale(2, RoundingMode.HALF_UP).doubleValue();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task03.java:46]:
```suggestionquantityMarks
            averageMarks[i] = BigDecimal.valueOf(summ / marks[i].length).setScale(2, RoundingMode.HALF_UP).doubleValue();
```

File: src/main/java/com/rakovets/course/java/core/practice/array/Task03.java [hunk 5]
```diff
@@ -45,9 +55,17 @@ public static void main(String[] args) {
      * @return минимальная отметка
      */
     static int[] getMinMarks(int[][] marks) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        int[] minimalMarks = new int[marks.length];
+        for (int i = 0; i < marks.length; i++) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/array/Task03.java:59]: Best way: using `foreach`

**Review by rakovets — COMMENTED** [review]
You need to fix the following problems

**Comment by davydovbv** [comment]
have no idea how to use foreach in the 3rd task.

**Review by rakovets — APPROVED** [review]
Approved

**Pull request closed without merging**